### ISSUE TITLE
Code run with cProfile performance regression in 3.11

### ISSUE THREAD

**Issue by Fidget-Spinner** [python/cpython#93381]
See this discussion on [Discourse](https://discuss.python.org/t/project-run-on-python-3-11-0b1-is-1-6x-slower-than-python-3-10/16067/3).

~We could speed up cProfile a few ways. I'll try to work on a few PRs.~ We first need to find out why cProfile is so slow. I'm not sure if we can backport fixes. Though a 1.6x slowdown seems pretty significant.

**Comment by Fidget-Spinner** [comment]
An update: I can reproduce the results that OP posted for his application. That is:
```
3.10 with cProfile:
Time Elapsed: 3.6727186999924015

3.11 with cProfile:
Time Elapsed: 6.189334200011217
```

Without cProfile, there's a 10-20% speedup over 3.10.
```
python main.py -n 500

3.10 without cProfile:
Time Elapsed: 15.240909200001624

3.11 without cProfile:
Time Elapsed: 12.682664599997224
```

**Comment by Fidget-Spinner** [comment]
Another update: between python 3.10 and 3.11, `test_cprofile` didn't slow down at all (in fact, it sped up slightly). That likely indicates that cprofile itself didn't slow down, it's likely the impact of tracing code went up instead.

**Comment by Fidget-Spinner** [comment]
This was mostly fixed in #93516. The next release (3.11b4) will have all the fixes included.
```
3.10 profiler.py
Time Elapsed: 2.1907681000002412

3.11b4 profiler.py
Time Elapsed: 2.4747707999995328
```
